src/Persisters/Collection/ManyToManyPersister.php

```diff
@@ -249,7 +253,7 @@ public function loadCriteria(PersistentCollection $collection, Criteria $criteri
                 $whereClauses[] = sprintf('te.%s %s NULL', $field, $operator === Comparison::EQ ? 'IS' : 'IS NOT');
             } else {
                 $whereClauses[] = sprintf('te.%s %s ?', $field, $operator);
-                $params[]       = $value;
+                $params         = array_merge($params, $this->getValues($value));
```

Comment:
Please don't `array_merge()` in a foreach loop. Create an array of arrays and perform a big merge in the end.

Reply:
All right. So I remove also `array_merge` calls in foreach loops in the `ResolveValuesHelper` though